test/exmg-form.spec.ts

```diff
@@ -0,0 +1,68 @@
+import {ExmgForm} from '../exmg-form';
+import {promisifyFlush, onExmgFormCancel, onExmgFormSubmit} from './utils';
+import {PaperInputElement} from "@polymer/paper-input/paper-input";
+
+declare const fixture: <T extends HTMLElement = HTMLElement>(id: string, model?: object) => T;
+declare const flush: (cb?: Function) => void;
+
+const {assert} = chai;
+
+suite('<exmg-form>', function () {
+  let element: ExmgForm;
+  const flushCompleted = promisifyFlush(flush);
+
+  suite('base usage', function () {
+    setup(() => {
+      element = fixture('ExmgFormBasicElement');
+    });
+
+    test('element is upgraded', function () {
+      assert.instanceOf(element, ExmgForm);
+    });
+
+    test('submitting form', async () => {
+      await flushCompleted();
+
+      const field1Input = <PaperInputElement>element.querySelector('paper-input[name=field1]');
+      const submitBtn = <HTMLElement>element.shadowRoot!.querySelector('paper-button.primary');
+
+      const eventPromise = onExmgFormSubmit(element, true);
+
+      field1Input.value = 'test1';
+      submitBtn.click();
+
+      const {detail} = await eventPromise;
+
+      assert.equal(detail.field1, 'test1');
+      assert.equal(detail.field2, 'test2');
+    });
+
+    test('form with missing required fields should not submit data', async () => {
+      await flushCompleted();
+      const submitBtn = <HTMLElement>element.shadowRoot!.querySelector('paper-button.primary');
+
+      const eventPromise = onExmgFormSubmit(element, false);
+
+      submitBtn.click();
+
+      const timeoutPromise = new Promise((resolve) => {
+        setTimeout(() => {
+          resolve();
+        }, 1000);
+      });
+
+      return await Promise.race([timeoutPromise, eventPromise]);
```

Comment:
!Not really sure what you are testing here. This test will be always valid regardless of form submitting.